ArduPlane/Log.cpp

```diff
@@ -159,6 +159,61 @@ struct PACKED log_Status {
     bool impact;
 };
 
+// precision landing logging
```

Comment:
If this logging is unchaged from Copter it should be moved to within the precland library.

Reply:
I'll take a look at it.
I removed precision loiter because it wasn't a feature that the company needed. I could add it but I dont have a bird to test that feature on since it is outside the scope of what was asked of me.

Reply:
So if I do that refactor will I need to change it in copter as well?

Reply:
I'm not sure what you mane by moving it to the Precland library.
I see a couple options here:
1. Logging is totally internal to precland
2. Logging functions are within precland but are called by the logging functions of whoever is using it.
Does AC_Precland need its own AC_Logging object and that object will then get initiated and run its operations?
I am not as familiar with this part of the architecture, any help here would be great.

Reply:
Making it entirely internal to PrecLand would make it very similar to what we do in TECS: https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_TECS/AP_TECS.cpp#L1080

I'm not sure exactly what the best logging of PrecLand would be, but for the time being you would probably emulate what we've got in Copter at the moment - just log based on wall-clock time.  The shape of this is usually to have a member variables `const uint16_t log_interval_ms = 40; # 25Hz logging` and `uint32_t last_log_time_ms;` in the precland instance, and have code something like this in `AC_PrecLand::update()`:

```
const uint32_t now = AP_HAL::millis();
if (now - last_log_time > log_interval) {
  last_log_time = now;
  log_PREC_message();
}
```

where the body of log_PREC_message() looks much like the code in TECS, using `Log_Write` (tridge suggested this on the devcall; I suggested we do it otherwise, but on review I was wrong on the devcall).

This cleanup could very easily be done as a precursor PR to what you have - let me know if you'd like me to take this on, but you might find it an interesting thing to do.  (If so, while you're there you could nuke the initialisation of the Vector objects - that just wastes space as they're one of the few objects we implicitly zero in their constructor, so initialisation just wastes flash).